### PR TITLE
[16.0][ADD] purchase_report_date_format

### PR DESCRIPTION
**Pull request by rinaldifirdaus**
This module shows following datetime fields of purchase reports in date format instead of datetime,
since presenting dates up to seconds is too much or unnecessary in many purchase
transactions.

Request for Quotation print:

* Expected Date

Purchase Order print:

* Order Date
* Requested Date

@qrtl

### CODE REVIEW

**Comment by yostashiro** [comment]
@rinaldifirdaus Can you please create another PR to fix the pre-commit issue, running copier update.

**Comment by yostashiro** [comment]
@rinaldifirdaus Can we take care of 'Order Deadline' presentation as well?

image

File: purchase_report_date_format/reports/purchase_order_templates.xml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+<odoo>
+    <template
+        id="report_purchaseorder_document_inherit"
+        inherit_id="purchase.report_purchaseorder_document"
+    >
+        <xpath expr="//p[@t-field='o.date_approve']" position="attributes">
+            <attribute name="t-att-style">'display: none'</attribute>
+        </xpath>
+        <xpath expr="//p[@t-field='o.date_approve']" position="after">
+            <p t-field="o.date_approve" class="m-0" t-options='{"widget": "date"}' />
```
Comment (youring) [inline purchase_report_date_format/reports/purchase_order_templates.xml:11]:
```suggestion
        <xpath expr="//p[@t-field='o.date_approve']" position="attributes">
            <attribute name="t-options">{&quot;widget&quot;: &quot;date&quot;}</attribute>
        </xpath>
```

Comment (youring) [inline purchase_report_date_format/reports/purchase_order_templates.xml]:
```suggestion
            <attribute name="t-options">{"widget": "date"}</attribute>
```

Comment (youring) [reply purchase_report_date_format/reports/purchase_order_templates.xml]: Maybe direct mod on t-options attribute is better than hide and re-add the same field. Same below.

File: purchase_report_date_format/reports/purchase_order_templates.xml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+        </xpath>
+        <xpath expr="//span[@t-field='line.date_planned']" position="attributes">
+            <attribute name="t-att-style">'display: none'</attribute>
+        </xpath>
+        <xpath expr="//span[@t-field='line.date_planned']" position="after">
+            <span t-field="line.date_planned" t-options='{"widget": "date"}' />
```
Comment (youring) [inline purchase_report_date_format/reports/purchase_order_templates.xml]:
```suggestion
            <attribute name="t-options">{"widget": "date"}</attribute>
```

File: purchase_report_date_format/reports/purchase_quotation_templates.xml [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+<odoo>
+    <template
+        id="report_purchasequotation_document_inherit"
+        inherit_id="purchase.report_purchasequotation_document"
+    >
+        <xpath expr="//span[@t-field='order_line.date_planned']" position="attributes">
+            <attribute name="t-att-style">'display: none'</attribute>
+        </xpath>
+        <xpath expr="//span[@t-field='order_line.date_planned']" position="after">
+            <span t-field="order_line.date_planned" t-options='{"widget": "date"}' />
```
Comment (youring) [inline purchase_report_date_format/reports/purchase_quotation_templates.xml]:
```suggestion
            <attribute name="t-options">{"widget": "date"}</attribute>
```

File: purchase_report_date_format/reports/purchase_order_templates.xml (hunk 1 above)
Comment (rinaldifirdaus) [reply purchase_report_date_format/reports/purchase_order_templates.xml]: Thanks for your suggestions!

File: purchase_report_date_format/readme/DESCRIPTION.rst [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+since presenting dates up to seconds is too much or unnecessary in many purchase
+transactions.
+
+Request for Quotation print:
+
+* Expected Date
+
+Purchase Order print:
+
+* Order Date
```
Comment (yostashiro) [inline purchase_report_date_format/readme/DESCRIPTION.rst:11]:
```suggestion
* Order Date
* Order Deadline
```

**Review by yostashiro — APPROVED** [review]
@rinaldifirdaus Please squash your commits.

**Review by AungKoKoLin1997 — APPROVED** [review]
LGTM

**Pull request merged**